cmd/controller-manager/app/controllermanager.go

```diff
@@ -564,12 +557,14 @@ func setupControllers(mgr controllerruntime.Manager, opts *options.Options, stop
 		RESTMapper:                      mgr.GetRESTMapper(),
 		DynamicClient:                   dynamicClientSet,
 		SkippedResourceConfig:           skippedResourceConfig,
-		SkippedPropagatingNamespaces:    skippedPropagatingNamespaces,
 		ResourceInterpreter:             resourceInterpreter,
 		EventRecorder:                   mgr.GetEventRecorderFor("resource-detector"),
 		ConcurrentResourceTemplateSyncs: opts.ConcurrentResourceTemplateSyncs,
 		RateLimiterOptions:              opts.RateLimiterOpts,
 	}
+	for _, ns := range opts.SkippedPropagatingNamespaces {
+		resourceDetector.SkippedPropagatingNamespaces = append(resourceDetector.SkippedPropagatingNamespaces, regexp.MustCompile(fmt.Sprintf("^%s$", ns)))
+	}
```

Comment:
Can we have a method on [type Options struct](https://github.com/karmada-io/karmada/blob/9d898fd4e6caa016d9fc23b314a354c75668563a/cmd/controller-manager/app/options/options.go#L31)? like:
```golang
func (o *Options) SkippedNamespacesRegexps() []*regexp.Regexp
```
For duplicate code concerns.